encrypted-media-respec.html

```diff
@@ -1598,6 +1591,10 @@
         The above rule implies that the CDM instance must not be destroyed until all <a>MediaKeys</a>
         objects and all <a>MediaKeySession</a> objects associated with the CDM instance are destroyed.
       </p>
+      <p>
+        When a <a>MediaKeySession</a> object is destroyed, the CDM SHALL close the <a href="#key-session">key session</a> associated with the object.
```

Comment:
The reason to use "object becoming inaccessible" is that this is an event that is unambiguous across browsers. There is a point in time at which the last reference from Javascript to the object is dropped and this is the same point in time for every implementation, because there is a specific Javascript instruction in the page that causes the last reference to be dropped.

So, the point is to say that as a consequence of this happening, the CDM session will be closed.

The previous text did say "if ...". This is a straightforward logical requirement that doesn't specify any timing (so it does not imply "when"). But it does imply that closure of the session happens _after_ the object becomes inaccessible because effects always come after causes :-)